pkg/container-collection/options.go

```diff
@@ -266,6 +283,7 @@ func WithInitialKubernetesContainers(nodeName string) ContainerCollectionOption
 		if err != nil {
 			return fmt.Errorf("failed to create Kubernetes client: %w", err)
 		}
+		defer k8sClient.Close()
```

Comment:
I am not sure to understand.
This functions is already in `cc.cleanUpFuncs`, so it will be called twice?
Should it be removed from the array?

Reply:
Those invocations are for different objects. `k8sClient.Close()` on L195 closes the object created on L181. This invocation on L286 closes the object created on L282.